src/webgpu/api/operation/storage_texture/read_only.spec.ts

```diff
@@ -0,0 +1,378 @@
+        return new Uint32Array(arrayBuffer);
+      case 'rgba8uint':
+      case 'rgba8unorm':
+        return new Uint8Array(arrayBuffer);
+      case 'rgba16uint':
+        return new Uint16Array(arrayBuffer);
+      case 'r32sint':
+      case 'rg32sint':
+      case 'rgba32sint':
+        return new Int32Array(arrayBuffer);
+      case 'rgba8sint':
+      case 'rgba8snorm':
+        return new Int8Array(arrayBuffer);
+      case 'rgba16sint':
+        return new Int16Array(arrayBuffer);
+      case 'r32float':
+      case 'rg32float':
+      case 'rgba32float':
+        return new Float32Array(arrayBuffer);
+      case 'rgba16float':
+        return new Float16Array(arrayBuffer);
+      default:
+        unreachable();
+        return new Uint8Array(arrayBuffer);
+    }
+  }
+
+  GetOutputBufferWGSLType(format: GPUTextureFormat) {
+    switch (format) {
+      case 'r32uint':
+      case 'rg32uint':
+      case 'rgba32uint':
+      case 'rgba8uint':
+      case 'rgba16uint':
+        return 'vec4u';
+      case 'r32sint':
+      case 'rg32sint':
+      case 'rgba8sint':
+      case 'rgba16sint':
+      case 'rgba32sint':
+        return 'vec4i';
+      case 'r32float':
+      case 'rg32float':
+      case 'rgba32float':
+      case 'rgba8snorm':
+      case 'rgba8unorm':
+      case 'rgba16float':
+        return 'vec4f';
+      default:
+        unreachable();
+        return '';
+    }
+  }
+
+  DoTransform(storageTexture: GPUTexture, outputBuffer: GPUBuffer) {
+    const declaration =
+      storageTexture.depthOrArrayLayers > 1 ? 'texture_storage_2d_array' : 'texture_storage_2d';
+    const textureDeclaration = `
+    @group(0) @binding(0) var readOnlyTexture: ${declaration}<${storageTexture.format}, read>;
+    `;
+
+    const textureLoadCoord =
+      storageTexture.depthOrArrayLayers > 1
+        ? `vec2u(invocationID.x, invocationID.y), invocationID.z`
+        : `vec2u(invocationID.x, invocationID.y)`;
+    const computeShader = `
+    ${textureDeclaration}
+    @group(0) @binding(1)
+    var<storage,read_write> outputBuffer : array<${this.GetOutputBufferWGSLType(
+      storageTexture.format
+    )}>;
+    @compute
+    @workgroup_size(${storageTexture.width}, ${storageTexture.height}, ${
+      storageTexture.depthOrArrayLayers
+    })
+    fn main(
+      @builtin(local_invocation_id) invocationID: vec3u,
+      @builtin(local_invocation_index) invocationIndex: u32) {
+      let initialValue = textureLoad(readOnlyTexture, ${textureLoadCoord});
+      outputBuffer[invocationIndex] = initialValue;
+    }`;
+    const computePipeline = this.device.createComputePipeline({
+      compute: {
+        module: this.device.createShaderModule({
+          code: computeShader,
+        }),
+      },
+      layout: 'auto',
+    });
+    const bindGroup = this.device.createBindGroup({
+      layout: computePipeline.getBindGroupLayout(0),
+      entries: [
+        {
+          binding: 0,
+          resource: storageTexture.createView(),
```

Comment:
nit: in the future it would be nice to test at a mip level as well. Not blocking for shipping IMHO